genshin/client/components/base.py

```diff
@@ -307,6 +316,13 @@ async def _request_hook(
             params = {k: v for k, v in params.items() if k != "authkey"}
             url = url.update_query(params)
 
+        headers_ = dict(headers or {})
+        if self.device_id:
+            headers_["x-rpc-device_id"] = self.device_id
+        if self.device_fp:
+            headers_["x-rpc-device_fp"] = self.device_fp
+        headers.update(headers_)
```

Comment:
Could you maybe do this instead?
```python
headers.update({k: v for k, v in {
    "x-rpc-device_id": self.device_id,
    "x-rpc-device_fp": self.device_fp
}.items() if v})
```

Reply:
I kinda like the previous one more, there was no need for a `headers_` variable though.

Reply:
Would be better to avoid more than one if that serves the same purpose